packages/legend-application-studio/src/components/editor/side-bar/testable/GlobalTestRunner.tsx

```diff
@@ -417,68 +452,117 @@ export const GlobalTestRunner = observer(
-              tabIndex={-1}
-              title="Run All Tests"
-            >
-              <PlayIcon />
-            </button>
-          </div>
-        </div>
-        <div className="panel__content side-bar__content">
-          <PanelLoadingIndicator isLoading={isDispatchingAction} />
-          <div className="panel side-bar__panel">
-            <div className="panel__header">
-              <div className="panel__header__title">
-                <div className="panel__header__title__content">TESTABLES</div>
+
+    const renderTestRunnerTab = (): React.ReactNode => {
+      if (
+        selectedTab === GLOBAL_TEST_RUNNER_TABS.GLOBAL_TEST_RUNNER.valueOf()
+      ) {
+        return (
+          <div
+            data-testid={LEGEND_STUDIO_TEST_ID.GLOBAL_TEST_RUNNER}
+            className="panel global-test-runner"
+          >
+            <PanelHeader className="panel__header side-bar__header">
+              <div className="panel__header__title global-test-runner__header__title">
+                <div className="panel__header__title__content side-bar__header__title__content">
+                  GLOBAL TEST RUNNER
+                </div>
               </div>
+              <div className="panel__header__actions side-bar__header__actions">
+                <button
+                  className={clsx(
+                    'panel__header__action side-bar__header__action global-test-runner__refresh-btn',
+                    {
+                      'global-test-runner__refresh-btn--loading':
+                        isDispatchingAction,
+                    },
+                  )}
+                  disabled={isDispatchingAction}
+                  onClick={reset}
+                  tabIndex={-1}
+                  title="Run All Tests"
+                >
+                  <RefreshIcon />
+                </button>
+                <button
+                  className="panel__header__action side-bar__header__action global-test-runner__refresh-btn"
+                  disabled={isDispatchingAction}
+                  onClick={runAllTests}
+                  tabIndex={-1}
+                  title="Run All Tests"
+                >
+                  <PlayIcon />
+                </button>
+              </div>
+            </PanelHeader>
+            <PanelContent className="side-bar__content">
+              <PanelLoadingIndicator isLoading={isDispatchingAction} />
+              <Panel className="side-bar__panel">
+                <PanelHeader>
+                  <div className="panel__header__title">
+                    <div className="panel__header__title__content">
+                      TESTABLES
+                    </div>
+                  </div>
+                  <div
+                    className="side-bar__panel__header__changes-count"
+                    data-testid={
+                      LEGEND_STUDIO_TEST_ID.SIDEBAR_PANEL_HEADER__CHANGES_COUNT
+                    }
+                  ></div>
```

Comment:
Is there ever a time that something gets written in this `LEGEND_STUDIO_TEST_ID.SIDEBAR_PANEL_HEADER__CHANGES_COUNT` div?

Reply:
Ok added count of `globalTestRunnerState.testableStates` which is hopefully the one you wanted 🙇